polaris-react/src/components/Picker/Picker.tsx

```diff
@@ -0,0 +1,265 @@
+    if (shouldOpen) {
+      handleOpen();
+    }
+  }, [shouldOpen, handleOpen]);
+
+  const handleBlur = useCallback(() => {
+    if (popoverActive) {
+      handleClose();
+    }
+  }, [popoverActive, handleClose]);
+
+  const textFieldContextValue: ComboboxTextFieldType = useMemo(
+    () => ({
+      activeOptionId,
+      expanded: popoverActive,
+      listboxId,
+      setTextFieldFocused,
+      setTextFieldLabelId,
+      onTextFieldFocus: handleFocus,
+      onTextFieldChange: handleChange,
+      onTextFieldBlur: handleBlur,
+    }),
+    [
+      activeOptionId,
+      popoverActive,
+      listboxId,
+      setTextFieldFocused,
+      setTextFieldLabelId,
+      handleFocus,
+      handleChange,
+      handleBlur,
+    ],
+  );
+
+  const listboxOptionContextValue: ComboboxListboxOptionType = useMemo(
+    () => ({
+      allowMultiple,
+    }),
+    [allowMultiple],
+  );
+
+  const listboxContextValue: ComboboxListboxType = useMemo(
+    () => ({
+      listboxId,
+      textFieldLabelId,
+      textFieldFocused,
+      willLoadMoreOptions,
+      setActiveOptionId,
+      setListboxId,
+      onKeyToBottom: onScrolledToBottom,
+    }),
+    [
+      listboxId,
+      textFieldLabelId,
+      textFieldFocused,
+      willLoadMoreOptions,
+      setActiveOptionId,
+      setListboxId,
+      onScrolledToBottom,
+    ],
+  );
+
+  const updateText = useCallback(
+    (value: string) => {
+      setQuery(value);
+
+      if (value === '') {
+        setFilteredOptions(options);
+        return;
+      }
+
+      const resultOptions = options?.filter((option) =>
+        FILTER_REGEX(value).exec(reactChildrenText(option.children)),
+      );
+      setFilteredOptions(resultOptions ?? []);
+    },
+    [options],
+  );
+
+  const firstSelectedOption = reactChildrenText(
+    options.find((option) => option.value === activeItem)?.children,
+  );
+
+  const firstSelectedLabel = firstSelectedOption
+    ? firstSelectedOption?.toString()
+    : activator.placeholder;
+
+  const queryMatchesExistingOption = options.some((option) =>
+    QUERY_REGEX(query).exec(reactChildrenText(option.children)),
+  );
+
+  return (
+    <Popover
+      active={popoverActive}
+      activator={
+        <Activator
+          {...activator}
+          onClick={handleOpen}
+          placeholder={firstSelectedLabel}
+        />
+      }
+      autofocusTarget="none"
+      preventFocusOnClose
+      fullWidth
+      preferInputActivator={false}
+      preferredPosition="cover"
+      onClose={handleClose}
+    >
+      <Popover.Pane onScrolledToBottom={onScrolledToBottom} height={height}>
+        {searchField ? (
+          <Box
+            paddingBlockStart="200"
+            paddingBlockEnd="100"
+            paddingInline="200"
+            borderBlockEndWidth="025"
+            borderColor="border"
+          >
+            <ComboboxTextFieldContext.Provider value={textFieldContextValue}>
+              <SearchField
```

Comment:
I've noticed this as well and in testing. It's from calling focus on a ref and I haven't figure out a way to get by it. Using useEffect causing scroll to top issues and same with autoFocus. Might need your help jamming on this one

Reply:
I've tried to debug this a bit locally. When I remove the `textFieldFocus` and `setTextFieldFocus` wherever they're defined/used in Picker, the error never shows and SearchField behavior still remained the same. The issue with that is we lose keyboard nav on the Listbox. But if I made the `textFieldFocused` context on Listbox tied to `popoverActive`, keyboard nav works as expected. 

Since `focused` on SearchField is always set to true, I think this would work but also very happy to have a jam session to dive deeper and verify if there's any ramifications to this. I'm not 100% sure what the scroll to top issues you were having were but testing locally, I can still scroll on the overall DetailsPage regardless of whether Picker is open or not.

Reply:
This should work! Thanks Lo!